webapp/channels/src/packages/mattermost-redux/src/reducers/entities/users.ts

```diff
@@ -505,9 +505,17 @@ function dndEndTimes(state: RelationOneToOne<UserProfile, number> = {}, action:
         return addToState(state, userId, endTime);
     }
     case UserTypes.RECEIVED_STATUSES: {
-        const userStatuses: UserStatus[] = action.data;
+        const nextState = Object.assign({}, state);
```

Comment:
Let's also ensure to add a comment explaining why we are doing it this way. And also, this is why I raised the topic of CI because it is easy for a new dev to add more code using `.reduce` not realizing it can cause performance issues. But it's out of scope for now.

Reply:
this is now changed